nightfall-optimist/src/services/transaction-sign-send.mjs

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-param-reassign */
+
+import config from 'config';
+import { web3 } from '@polygon-nightfall/common-files/utils/contract.mjs';
+import { estimateGas, estimateGasPrice } from '@polygon-nightfall/common-files/utils/gas.mjs';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import { Mutex } from 'async-mutex';
+
+const { GAS, GAS_MULTIPLIER, GAS_ESTIMATE_ENDPOINT, GAS_PRICE, GAS_PRICE_MULTIPLIER } = config;
+
+const nonceMutex = new Mutex();
+
+const isListening = async () => {
+  try {
+    await web3.eth.net.isListening();
+  } catch (error) {
+    throw new Error('Web3 ws is disconnected, try again later');
```

Comment:
We could get the message from the `error` variable and pass in the exception.

Reply:
Tbh I think this implementation is a bit off, I'll take another look :pray:

Reply:
FYI `isListening` can return a boolean but also fail for many reasons
For the happy path, I am now returning the result from calling the web3 method, then I only proceed with the transaction if the result is "truthy"
If there's an error, the exception will be raised re-using the error message as per your suggestion